### DISCUSSION TITLE
fix-avsync.lua这个lua修复的问题是不是在mpv中已经被修复了

### DISCUSSION THREAD

**Discussion by wli189** [dyphire/mpv-config#80] [Q&A] [answered]
在使用embyToLocalPlayer和tsukimi调用mpv播放时，不会直接播放，好像是和这个lua冲突了

**Comment by dyphire** [comment, marked as answer]
尽管原始 issue https://github.com/mpv-player/mpv/issues/9591 已被关闭，但该问题并未真正修复。复现方式：加载带有高码率外部音轨（通常为 truehd）的视频文件，在播放过程中切换至外部高码率音轨，视频将持续卡顿直至十秒或是数十秒后再次同步

fix-avsync.lua 通过在切换音轨时执行帧步进帧步退的方式修复这一问题。不会自动播放很可能是你设置了以暂停状态启动 mpv 或调用软件传递了暂停状态给 mpv

**Reply by wli189** [reply]
如果是以播放列表传递呢
pub fn playlist_load_list(&self, path: &str, replace: bool) -> Result<()>

**Reply by dyphire** [reply]
关键不在于传递文件的方式，而在于有没有传递 pause 状态，fix-avsync.lua 只有在原本就是 pause 时才不会在执行后取消 pause 状态

**Reply by wli189** [reply]
貌似如果传递了播放进度就pause了

**Reply by dyphire** [reply]
你能通过播放 youtube 的进度链接复现这一点吗？win 电脑不在身边。可以考虑把协议播放排除在脚本之外，它们不太可能触发这个问题，由于我基本不播放网络协议文件，所以之前没考虑到这点

**Reply by dyphire** [reply]
更新脚本：https://github.com/dyphire/mpv-config/commit/7a4caeed7280b9d1fb38b2c8dff138ba51f8e2a2 ，重新测试下看看

**Reply by wli189** [reply]
好了，应该和链接播放有点小关系。感谢
但看了下log
[2583.367][f][fix_avsync] Lua error: C:\Users\brian\AppData\Roaming/mpv/scripts/fix-avsync.lua:14: ')' expected near '�'
，是中文的逗号

**Reply by dyphire** [reply]
local path = mp.get_property("path", "")
输入法的锅，回去再改了，mac 上没关联 git 账号

**Comment by dyphire** [comment]
切换外部音轨导致的 A-V 同步问题已在上游提交中修复：https://github.com/mpv-player/mpv/commit/940854c ，之后的 mpv 版本不再需要 fix-avsync.lua 脚本
